src/main/java/com/flightstats/hub/dao/aws/S3Config.java

```diff
@@ -92,10 +93,10 @@ protected Scheduler scheduler() {
 
     }
 
-    private class S3ConfigLockable implements Lockable {
+    class S3ConfigLockable implements Lockable {
```

Comment:
Ah, this was an existing thing and you were trying to change as little as possible. I think I'd extract all the max update methods into a top-level class and use it as a dependency in this lockable and its parent. Then this can go back to hiding as private. Does that make sense?

Reply:
makes sense, shied away from changing too much in the first iteration but changing this was easy without breaking anything